spikingjelly/activation_based/triton_kernel/flex_sn_inductor/kernel.py

```
@@ -1,205 +1,267 @@
-
-
-def build_training_kernels(
-    core_fn: Callable,
-    num_inputs: int,
-    num_states: int,
-    num_outputs: int,
-    example_inputs: Optional[Tuple[torch.Tensor, ...]] = None,
-):
-    """Build forward + backward Triton scan kernels for BPTT training.
+    safe_name = "".join(c if c.isalnum() else "_" for c in raw_name) or "core"
+    graph_fingerprint = hashlib.sha1(
```

Ruff (0.15.12)

[error] Probable use of insecure hash functions in `hashlib`: `sha1`

(S324)